### DISCUSSION TITLE
M5Stack Core2 Support

### DISCUSSION THREAD

**Discussion by LukePrior** [dl9rdz/rdz_ttgo_sonde#133] [Q&A]
I see in your latest pull request you seem to have added support for the M5Stack Core2, could you please describe the setup. Did you use the M5Stack Core2 and the LoRa board or something else?

**Comment by DL2MF** [comment]
I'm using this setup, it's the genuine M5STACK Core:

![grafik](https://user-images.githubusercontent.com/1955750/129457346-737aec88-7b4c-43cd-8579-5338d3ea10f8.png)

**Reply by LukePrior** [reply]
What are the two stacks?

**Reply by LukePrior** [reply]
![Screenshot_2021-08-15-10-57-32-42_e4424258c8b8649f6e67d283a50a2cbc](https://user-images.githubusercontent.com/22492406/129463698-7a416a2c-0f7f-4f75-9737-f7a455a97714.jpg)

**Comment by dl9rdz** [comment]
So my test setup uses the Core2 (which already includes a battery), and the LoRa Module (433), optionally also a GPS module (which is relatively expensive). It should be rather easy to adjust the code for the Basic Core, but a few pins are different (SX1278 MISO, possibly also the SPI-SS lines of Display and/or SX1278). Autodetection detects the Core2 by scanning the I2C bus for an BM8563 which probably only exists on the Core2.

**Reply by LukePrior** [reply]
> So my test setup uses the Core2 (which already includes a battery), and the LoRa Module (433), optionally also a GPS module (which is relatively expensive). It should be rather easy to adjust the code for the Basic Core, but a few pins are different (SX1278 MISO, possibly also the SPI-SS lines of Display and/or SX1278). Autodetection detects the Core2 by scanning the I2C bus for an BM8563 which probably only exists on the Core2.

Oh so you can use the LoRa Module in the module above with the Core2?
I might have to look into picking a set up seems like quite a good little kit.

**Comment by dl9rdz** [comment]
Its not perfect, but works well.

You have to remove the back cover of the Core2, which serves as a battery holder. Which means then that the battery remains in the Core2 freely floating around. A small piece of tape is fine to fix that. Then, the LoRa module fits perfectly. The only remaining issue is that you still need a back cover for the LoRa module (or the GPS module if thats the last module). Its just a 5x5cm piece of plastic, so easy to DIY, but the removed back cover of the Core2 will not fit on the LoRa or GPS module.

A nice and easy solution would be to 3D-print a bottom module which you can equip with a GPS module recycled from a recovered RS41 :-)

**Reply by LukePrior** [reply]
> You have to remove the back cover of the Core2, which serves as a battery holder.

Ok so having a look it seems like the M5GO Battery Bottom2 for the Core2 solves this issue. They also appear to have M5Stack Core2 ESP32 IoT Development Kit for AWS IoT EduKit for the same price as the regular Core2 which includes the battery bottom.

![image](https://user-images.githubusercontent.com/22492406/129473062-8c278a35-3b21-426c-a3d5-86e57bffde8d.png)

**Reply by Mooni123** [reply]
How about the button functionality ?
The M5Stack Core2 has a touchscreen with 3 usable softbuttons.
So we need touchscreen support in RDZ_(TTGo)_Sonde ?!
